torch/csrc/autograd/input_metadata.h

```diff
@@ -67,9 +105,26 @@ struct InputMetadata {
     return at::zeros(shape_, options_);
   }
 
```

Comment:
Saw c10:optional floating around wonder if that is better replacement for the null values